ppb/colors.py

```diff
@@ -0,0 +1,36 @@
+from dataclasses import dataclass
+import colorsys
+
+@dataclass()
+class Color():
+    """An RGB color, with red, green, and blue values ranging from 0-255."""
+    red: int
+    green: int
+    blue: int
+
+    def __post_init__(self):
+        for key, value in {'red': self.red, 'green': self.green, 'blue': self.blue}.items():
+            if value < 0:
+                raise ValueError(f'{key} cannot be less than 0.')
+            elif value > 255:
+                raise ValueError(f'{key} cannot be greater than 255.')
+    
+    def __iter__(self):
+        return (self.red, self.green, self.blue)
+
+    @staticmethod
+    def from_hsv(hue: float, saturation: float, value: float):
```

Comment:
I would rather have `HSV().to_rgb()`?

@pathunstrom your thoughts?

Reply:
Concerned about that approach - I think there should only be one class for color so that it is intuitive to users that that’s the class they should plug in whenever they need to use a color. Don’t want people to accidentally use an HSV color when the class is expecting an RGB color, for example.

Reply:
I decided to refactor it into an abstract base class `Color`, with `RGBColor` and `HSVColor` subclasses that all have a `to_rgb()` method that should generally be abstracted away from the user. Let me know what you think!